src/test/java/com/microsoft/jenkins/azuread/integrations/casc/ConfigAsCodeTest.java

```diff
@@ -0,0 +1,90 @@
+package com.microsoft.jenkins.azuread.integrations.casc;
+
+import com.microsoft.jenkins.azuread.AzureAdMatrixAuthorizationStrategy;
+import com.microsoft.jenkins.azuread.AzureSecurityRealm;
+import hudson.model.Item;
+import hudson.security.AuthorizationStrategy;
+import hudson.security.SecurityRealm;
+import io.jenkins.plugins.casc.ConfigurationContext;
+import io.jenkins.plugins.casc.Configurator;
+import io.jenkins.plugins.casc.ConfiguratorRegistry;
+import io.jenkins.plugins.casc.misc.ConfiguredWithCode;
+import io.jenkins.plugins.casc.misc.JenkinsConfiguredWithCodeRule;
+import io.jenkins.plugins.casc.model.CNode;
+import io.jenkins.plugins.casc.model.Mapping;
+import io.jenkins.plugins.casc.model.Scalar;
+import jenkins.model.Jenkins;
+import org.junit.Rule;
+import org.junit.Test;
+import org.jvnet.hudson.test.LoggerRule;
+import org.jvnet.hudson.test.recipes.LocalData;
+
+import java.util.List;
+import java.util.logging.Level;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertNull;
+import static org.junit.Assert.assertTrue;
+
+public class ConfigAsCodeTest {
+    private static final String TEST_UPN = "abc@jenkins.com";
+
+    @Rule
+    public JenkinsConfiguredWithCodeRule j = new JenkinsConfiguredWithCodeRule();
+
+    @Rule
+    public LoggerRule l = new LoggerRule().record(AzureAdMatrixAuthorizationStrategyConfigurator.class, Level.WARNING).capture(20);
+
+    @Test
+    @ConfiguredWithCode("configuration-as-code.yml")
+    public void should_support_configuration_as_code() {
+        SecurityRealm securityRealm = j.jenkins.getSecurityRealm();
+        assertTrue("security realm", securityRealm instanceof AzureSecurityRealm);
+        AzureSecurityRealm azureSecurityRealm = (AzureSecurityRealm) securityRealm;
+        assertEquals("clientId", azureSecurityRealm.getClientId());
+        assertEquals("clientSecret", azureSecurityRealm.getClientSecret());
```

Comment:
I think you need some changes on the getters around here, this shouldn't be returning a string but a secret, and please add a test that covers that it is exported as a secret and not plain text

Reply:
It indeed returns a String instead of a secret here. For testing secret thing, this plugin rewrite the "marshal/unmarshal" for xstream, does this affect casc?

Reply:
Um not sure...
1. Why does it do that?
2. Verify the client secret in the export test

Reply:
This is how encrypt the credentials in the `config.xml` file

Reply:
err.... why don't you just use the `Secret` type? Which will be done automatically for you?

Reply:
No idea, the code exists in the first commit......

Reply:
probably a good time to fix it
the getters look really strange for the client config

Reply:
no need to fix now, it only affects writing to xml file